js/NavMainComponent.js

```diff
@@ -100,6 +179,94 @@ NavMainComponent.prototype.closeSubNavs = function() {
+
+};
+
+/** Detect window height, adjust the number of items in the primary nav and check when to add "More" option
+ * Notes:
+ * - Feels like the JS should handle the creation of the third level menu to avoid unlessisary markup changes
+ * - no need for this to be on prototype
+ */
+
+NavMainComponent.prototype.adjustNavItems = function() {
+
+    var component = this,
+        availableHeight = component.$window.height(),
+        navItemsHeight = (component.$html.find('.nav-primary .nav-items').outerHeight(true) + component.$html.find('.jadu-branding').outerHeight(true)),
+        moreIconHeight = 72, // Pre calculated height of the "More" nav item
+        navItemsCountTotal = component.$html.find('.nav-primary .nav-items li').length,
+        numberOfHiddenNavItems = 0;
+
+    if (navItemsHeight + moreIconHeight > availableHeight) {
+        // If there is not enough space hide the last primary nav items
+        component.hidePrimaryNavItems(navItemsHeight, moreIconHeight, availableHeight);
+        // Get the number of hidden items to make only them visible in the tertiary menu
+        numberOfHiddenNavItems = component.$html.find('.nav-primary .nav-items li:hidden').length;
+        // Add "More" nav item and check its visibility if already exists
+        component.addMoreNavItem(numberOfHiddenNavItems);
+        // Hide the primary nav items duplicate in tertiary menu
+        component.hideMoreCategoriesTopItems(navItemsCountTotal, numberOfHiddenNavItems);
+    } else {
+        // Unhide items if they were hidden and there is space in the primary nav
+        component.$html.find('.nav-primary .nav-items li').show();
+        component.$html.find('.nav-primary .nav-items [label="More"]').hide();
+    }
+};
+
+NavMainComponent.prototype.hidePrimaryNavItems = function(navItemsHeight, moreIconHeight, availableHeight) {
+
+    var component = this,
+        nthChild = 2; // This number represents the item before the last in the nth-last-child;
+
+    while (navItemsHeight + moreIconHeight > availableHeight) {
+
+        // If last nav item is visible hide it
+        if (component.$html.find('.nav-primary .nav-items li:last-child').is(':visible')) {
+            component.$html.find('.nav-primary .nav-items li:last-child').hide();
+        } else {
+            // If last nav item is hidden hide the next one up
+            component.$html.find('.nav-primary .nav-items li:nth-last-child('+ nthChild +')').hide();
+            nthChild++;
+        }
+
+        // Recalculate nav items height based on items just hidden
+        navItemsHeight = (component.$html.find('.nav-primary .nav-items').outerHeight(true) + component.$html.find('.jadu-branding').outerHeight(true));
+    }
+};
+
+NavMainComponent.prototype.addMoreNavItem = function(numberOfHiddenNavItems) {
+
+    var component = this;
+
+    // Add the "More" nav item
+    if ((numberOfHiddenNavItems > 0) && (component.$html.find('.more-icon').length <= 0)){
```

Comment:
Does this need `component.$html.find('.more-icon').length <= 0` ? Can we have a negative length returned from a jQuery collection (I'm not sure)? To me, seems like this could be shortened to `!component.$html.find('.more-icon').length`